### DISCUSSION TITLE
Hosted Valhalla APIs?

### DISCUSSION THREAD

**Discussion by rtwalz** [valhalla/valhalla#3903] [General]
Hi all!

I'm a college student with a small side project, Routeshuffle.com, that generates random running routes. I use Valhalla for all sorts of things, to generate elevation graphs, directions, matrices, and isochrones. It is such a powerful tool and I'm really grateful that the library exists and for the contributors' work. I've been running Valhalla on a $60/month VPS, which has been working well. Although I am now questioning whether my small usage justifies running my own Valhalla instance. 

Are there any companies that run hosted Valhalla APIs? I'm aware of Stadia Maps, although with their recent price increases, I would need to pay $300/month to access the features I need.

Thanks!

**Comment by nilsnolde** [comment]
I guess your usage exceeds the demo server policy? I’m also aware of geoapify.com, they sometimes also fund bug fixes and they’re affordable. Isochrones will always be pretty limited unless you host yourself.

**Comment by nilsnolde** [comment]
Also, if your site is demonstrably not commercial and maybe the code is even open source, feel free to ask Stadia or Geoapify for a discount or even free usage of a certain plan. They might allow it, in worst case they turn you down, nothing lost:)

**Comment by ianthetechie** [comment]
Hey @rtwalz, Stadia Maps cofounder here ;) Send us an email and we'd be happy to talk!
